src/internal_events/splunk_hec.rs

```diff
@@ -44,3 +46,77 @@ impl InternalEvent for SplunkEventEncodeError {
         );
     }
 }
+
+#[cfg(feature = "sources-splunk_hec")]
+mod source {
+    use super::*;
+    use crate::sources::splunk_hec::ApiError;
+
+    #[derive(Debug)]
+    pub struct SplunkHECEventReceived;
+
+    impl InternalEvent for SplunkHECEventReceived {
+        fn emit_logs(&self) {
+            trace!(message = "received one event.");
+        }
+
+        fn emit_metrics(&self) {
+            counter!(
+                "events_processed", 1,
+                "component_kind" => "source",
+                "component_type" => "splunk_hec",
+            );
```

Comment:
Is there any information about the number of bytes handled?

Reply:
It's not easy to get, it would require either:
* Collecting all bytes on `gzip` [path](https://github.com/timberio/vector/blob/937b0edccaf2fa5b6f4aebcc1557fbf1ae6ee6f8/src/sources/splunk_hec.rs#L182), which would require additional allocation.
* Introduce a wrapper struct that implements `Read` and intercepts method calls when [decoding](https://github.com/timberio/vector/blob/937b0edccaf2fa5b6f4aebcc1557fbf1ae6ee6f8/src/sources/splunk_hec.rs#L371) to compute how many bytes were emitted in total. 

I don't see any other way.

Reply:
Ok, that sounds like much more effort than it's worth, at least until someone requests it.